src/d3a/models/powerflow/__init__.py

```diff
@@ -0,0 +1,123 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from abc import ABC, abstractmethod
+
+from d3a.models.strategy.load_hours import LoadHoursStrategy
+from d3a.models.strategy.storage import StorageStrategy
+from d3a.models.strategy.pv import PVStrategy
+from d3a.models.strategy.finite_power_plant import FinitePowerPlant
+from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+
+
+class PowerFlowBase(ABC):
+    def __init__(self, root_area, root_voltage):
+        """
+        :param root_area: contains tree structured hierarchical energy grid
+        :param root_voltage: contains the voltage of the top most hierarchy
+        """
+        self.root_voltage = root_voltage
+        self._grid_forming(root_area)
+
+    def _grid_forming(self, area):
```

Comment:
Very minor, but I would prefer to name this function _form_grid_from_area, if you prefer it like it is now feel free to leave it as is.